### DISCUSSION TITLE
installing on macOS

### DISCUSSION THREAD

**Discussion by greggjaskiewicz** [dotnet/maui#298] [Q&A]
Hi there, 

First of all, apologies if this is a bit of an off question. 
We are looking to start using Maui for upcoming project, with 2-3 year release timeline. 
I am completely new to Xamarin etc
I have installed Visual Studio on my mac. 

However, after checking out the code and trying to run ./build.sh - I get this error. 
I have tried installing the newer version of dotNet SDK, but it does not seem to be able to pick it up. I don't see any flags or such that I could pass into the build script to let it know where to find it. 

Are there any instructions out there that would allow me to get from 0 to getting this installed please?

```

/usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj]
Done Building Project "/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj" (_GenerateRestoreGraphProjectEntry target(s)) -- FAILED.
Done Building Project "/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target(s)) -- FAILED.

Build FAILED.

"/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target) (1) ->
(_FilterRestoreGraphProjectInputItems target) -> 
  /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/NuGet.targets(307,5): warning : Skipping restore for project '/Users/gj/Projects/maui/src/Platform.Handlers/samples/Sample.Droid/Sample.Droid.csproj'. The project file may be invalid or missing targets required for restore. [/Users/gj/Projects/maui/Xamarin.Forms.sln]
  /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/NuGet.targets(307,5): warning : Skipping restore for project '/Users/gj/Projects/maui/src/Platform.Renderers/src/Xamarin.Forms.Platform.Android.FormsViewGroup/Xamarin.Forms.Platform.Android.FormsViewGroup.csproj'. The project file may be invalid or missing targets required for restore. [/Users/gj/Projects/maui/Xamarin.Forms.sln]


"/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target) (1) ->
"/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj" (_GenerateRestoreGraphProjectEntry target) (14:4) ->
(_CheckForUnsupportedNETCoreVersion target) -> 
  /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj]

    2 Warning(s)
    1 Error(s)

Time Elapsed 00:00:05.60

```

Lastly, my dotnet --info
```

.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.0
 OS Platform: Darwin
 RID:         osx.11.0-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.101/

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  1.0.4 [/usr/local/share/dotnet/sdk]
  2.1.302 [/usr/local/share/dotnet/sdk]
  3.1.404 [/usr/local/share/dotnet/sdk]
  5.0.101 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
  
```
  
Cheers!

**Comment by mrwcjoughin** [comment]
Hi,

MAUI is a project in development by Microsoft with a release date of
November 2021.

If you are starting a project now just use Xamarin.Forms now and you can
upgrade to MAUI when it’s released.
MAUI is the successor to Xamarin.Forms so there will be a upgrade tool to
convert your project from one to the other when it is released.

On Wed, 06 Jan 2021 at 16:59, greg jaskiewicz <notifications@github.com>
wrote:

> Hi there,
>
> First of all, apologies if this is a bit of an off question.
> We are looking to start using Maui for upcoming project, with 2-3 year
> release timeline.
> I am completely new to Xamarin etc
> I have installed Visual Studio on my mac.
>
> However, after checking out the code and trying to run ./build.sh - I get
> this error.
> I have tried installing the newer version of dotNet SDK, but it does not
> seem to be able to pick it up. I don't see any flags or such that I could
> pass into the build script to let it know where to find it.
>
> Are there any instructions out there that would allow me to get from 0 to
> getting this installed please?
>
>
> /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj]
> Done Building Project "/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj" (_GenerateRestoreGraphProjectEntry target(s)) -- FAILED.
> Done Building Project "/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target(s)) -- FAILED.
>
> Build FAILED.
>
> "/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target) (1) ->
> (_FilterRestoreGraphProjectInputItems target) ->
>   /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/NuGet.targets(307,5): warning : Skipping restore for project '/Users/gj/Projects/maui/src/Platform.Handlers/samples/Sample.Droid/Sample.Droid.csproj'. The project file may be invalid or missing targets required for restore. [/Users/gj/Projects/maui/Xamarin.Forms.sln]
>   /Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin/NuGet.targets(307,5): warning : Skipping restore for project '/Users/gj/Projects/maui/src/Platform.Renderers/src/Xamarin.Forms.Platform.Android.FormsViewGroup/Xamarin.Forms.Platform.Android.FormsViewGroup.csproj'. The project file may be invalid or missing targets required for restore. [/Users/gj/Projects/maui/Xamarin.Forms.sln]
>
>
> "/Users/gj/Projects/maui/Xamarin.Forms.sln" (Restore target) (1) ->
> "/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj" (_GenerateRestoreGraphProjectEntry target) (14:4) ->
> (_CheckForUnsupportedNETCoreVersion target) ->
>   /usr/local/share/dotnet/sdk/2.1.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 3.1.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 3.1. [/Users/gj/Projects/maui/src/Platform.Handlers/tests/Xamarin.Platform.Handlers.UnitTests/Xamarin.Platform.Handlers.UnitTests.csproj]
>
>     2 Warning(s)
>     1 Error(s)
>
> Time Elapsed 00:00:05.60
>
>
> Lastly, my dotnet --info
>
>
> .NET SDK (reflecting any global.json):
>  Version:   5.0.101
>  Commit:    d05174dc5a
>
> Runtime Environment:
>  OS Name:     Mac OS X
>  OS Version:  11.0
>  OS Platform: Darwin
>  RID:         osx.11.0-x64
>  Base Path:   /usr/local/share/dotnet/sdk/5.0.101/
>
> Host (useful for support):
>   Version: 5.0.1
>   Commit:  b02e13abab
>
> .NET SDKs installed:
>   1.0.4 [/usr/local/share/dotnet/sdk]
>   2.1.302 [/usr/local/share/dotnet/sdk]
>   3.1.404 [/usr/local/share/dotnet/sdk]
>   5.0.101 [/usr/local/share/dotnet/sdk]
>
> .NET runtimes installed:
>   Microsoft.AspNetCore.All 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
>   Microsoft.AspNetCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.AspNetCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
>   Microsoft.NETCore.App 1.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 1.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>   Microsoft.NETCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
>
> To install additional .NET runtimes or SDKs:
>   https://aka.ms/dotnet-download
>
> Cheers!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/maui/discussions/298>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOGERQETP3XPRMIATZZMSTSYR3G3ANCNFSM4VXUUM4Q>
> .
>
-- 
Thanks,

Matthew Joughin
Founder / CTO

+27 81 529 7169
http://www.adaptableapps.net
